### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by wenweil**
testing
- added jest as a dependency
- create test for all message endpoints
- create test for many error scenarios
error
-create a new error code for empty message scenario

### CODE REVIEW

File: src/models/schemas.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+const account = {
```
Comment (wenhao-zhang) [inline src/models/schemas.js]: Can you remove this file for now? I don't think this is how we want to do this.

File: tests/Unit/message.test.js [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    //time: 1590021724501
+  };
+  const req = {
+    body,
+  };
+  const res = {
+    json: (ret) => {},
+  };
+  const next = (error) => {
+    hasError = true;
```
Comment (wenhao-zhang) [inline tests/Unit/message.test.js]: I don't think this check is necessary. You are already checking if the errors are equal. if you don't receive an error, next won't be called.

File: src/utils/error/errors.js [hunk 3]
```diff
@@ -38,6 +38,11 @@ errorInfo[(exports.MESSAGE_NOT_EXIST = "MessageIDNotExistError")] = {
   message: "Error: message with corresponding ID does not exist"
 };
 
+errorInfo[(exports.MESSAGE_NO_TEXT = "MessageNotContainText")] = {
```
Comment (wenhao-zhang) [inline src/utils/error/errors.js:41]: Should this be called MESSAGE_DOES_NOT_EXIST? I think we should use 404 for not found.

**Comment by wenhao-zhang** [comment]
Can you group the tests with describes?

File: tests/Unit/message.test.js [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+    const err = new Error(errors.PLEASE_AUTHENTICATE);
+    err.name = errors.PLEASE_AUTHENTICATE;
+    expect(error).toEqual(err);
+  };
+  await functions.newMessage(req, res, next);
+  expect(hasError).toBe(true);
+});
+test("new message no text", async () => {
```
Comment (wenhao-zhang) [inline tests/Unit/message.test.js]: I think this should use a different error from not found.

File: tests/Unit/message.test.js (hunk 2 above)
Comment (wenweil) [reply tests/Unit/message.test.js]: if next is not called, test passes. this test supposed to test if correct error is thrown

File: tests/Unit/message.test.js (hunk 4 above)
Comment (wenweil) [reply tests/Unit/message.test.js]: this is not not found. this is saying the message contains no text in its body for some reason

File: src/utils/error/errors.js (hunk 3 above)
Comment (wenweil) [reply src/utils/error/errors.js:41]: no this is really for a malformed request

Comment (wenweil) [reply src/utils/error/errors.js:41]: id probably should have put this somewhere in the validator

File: tests/Unit/message.test.js (hunk 2 above)
Comment (wenweil) [reply tests/Unit/message.test.js]: maybe i should put all of these test cases in a init thing, and then write a mock database

File: tests/Unit/message.test.js (hunk 4 above)
Comment (wenhao-zhang) [reply tests/Unit/message.test.js]: But you have a not found using the same error. You should create a new error for that one I think.

File: src/models/schemas.js (hunk 1 above)
Comment (wenweil) [reply src/models/schemas.js]: okay

File: tests/Unit/message.test.js (hunk 4 above)
Comment (wenweil) [reply tests/Unit/message.test.js]: this does not use not found, it uses bad request

File: tests/Unit/message.test.js [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+    const req = {
+      body,
+    };
+    const res = {
+      json: (ret) => {},
+    };
+    const next = (error) => {
+      hasError = true;
+      const err = new Error(errors.MESSAGE_NOT_EXIST);
+      err.name = errors.MESSAGE_NOT_EXIST;
```
Comment (wenhao-zhang) [inline tests/Unit/message.test.js:284]: This should use 404 instead of bad request

Comment (wenweil) [reply tests/Unit/message.test.js:284]: MESSAGE_NOT_EXIST is 404 tho
errorInfo[(exports.MESSAGE_NOT_EXIST = "MessageIDNotExistError")] = {
  status: httpStatus.NOT_FOUND,
  message: "Error: message with corresponding ID does not exist"
};

**Review by wenhao-zhang — APPROVED** [review]

**Pull request merged**